### PR TITLE
DirectoryLoader slicing

### PR DESCRIPTION
**Pull request by amovfx**
DirectoryLoader can now return a random sample of files in a directory.
Parameters added are:
sample_size
randomize_sample
sample_seed

@rlancemartin, @eyurtsev

### CODE REVIEW

**Comment by amovfx** [comment]
Found this helpful in prototyping in notebooks on large data sets.

File: libs/langchain/langchain/document_loaders/directory.py [hunk 1]
```diff
@@ -55,6 +60,8 @@ def __init__(
             show_progress: Whether to show a progress bar. Defaults to False.
             use_multithreading: Whether to use multithreading. Defaults to False.
             max_concurrency: The maximum number of threads to use. Defaults to 4.
+            sample_size: The maximum number of files you would like to load from the directory.
+            randomize_sample: Suffle the files to get a random sample.
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/directory.py:64]: Could you add documentation for sample_seed

File: libs/langchain/langchain/document_loaders/directory.py [hunk 2]
```diff
@@ -39,6 +41,9 @@ def __init__(
         show_progress: bool = False,
         use_multithreading: bool = False,
         max_concurrency: int = 4,
+        sample_size: Union[int, None] = None,
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/directory.py:44]: Could you add `*` here

```suggestion
        *,
        sample_size: Union[int, None] = None,
```

File: libs/langchain/langchain/document_loaders/directory.py (hunk 1 above)
Comment (amovfx) [reply libs/langchain/langchain/document_loaders/directory.py:64]: Sorry for missing that sure thing.

File: libs/langchain/langchain/document_loaders/directory.py (hunk 2 above)
Comment (amovfx) [reply libs/langchain/langchain/document_loaders/directory.py:44]: Will do. thanks for the review.

**Comment by baskaryan** [comment]
thanks @amovfx !

**Pull request merged**